src/events/messageCreate.ts

```diff
@@ -1,27 +1,57 @@
 import {
-	Message,
 	MessageOptions,
 	MessageActionRow,
 	MessageButton,
 	ThreadChannel,
+	TextChannel,
 } from 'discord.js';
 import { event } from 'jellycommands';
-import url_regex from 'url-regex-safe';
 import { AUTO_THREAD_CHANNELS } from '../config';
 import { wrap_in_embed } from '../utils/embed_helpers';
 import { add_thread_prefix } from '../utils/threads';
-import { get_title_from_url } from '../utils/unfurl';
-import { Url } from 'url';
 
 export default event({
 	name: 'messageCreate',
-	run: async ({}, message) => {
+	run: async ({ }, message) => {
+		// Rules for whether or not the message should be dealt with by the bot
 		const should_ignore =
 			message.author.bot ||
 			message.channel.type != 'GUILD_TEXT' ||
 			message.type != 'DEFAULT' ||
 			!AUTO_THREAD_CHANNELS.includes(message.channelId);
 
+		// If a response is sent by a user in an auto thread channel
+		if (message.type === 'REPLY' && AUTO_THREAD_CHANNELS.includes(message.channelId) && !message.author.bot) {
+			// Delete their message
+			await message.delete()
+			// Get the channel as a TextChannel
+			const channel = message.channel as TextChannel
+			// Get the thread that the message refers to
+			let thread = channel.threads.cache.get(message.reference.messageId) as ThreadChannel
+			// If the thread wasn't found in the cache
+			if (!thread) {
+				// Get all active threads in the guild
+				const threads = (await message.guild.channels.fetchActiveThreads()).threads
+				// Get the thread
+				thread = threads.get(message.reference.messageId) as ThreadChannel
+			}
+			// Make msg available outside the if/else scope
+			let msg;
+			// If the thread was found
+			if (thread) {
+				// Send the message to the thread
+				await thread.send(`> Response by <@${message.author.id}>\n` + message.content)
+				// Create response message to the user with a link to the thread
+				msg = wrap_in_embed(`Please send responses in the thread created for the issue and not directly in the thread channel.\n\nClick this link to go to the thread you should respond in\n<#${thread.id}>`)
+			} else {
+				// Create response message to the user
+				msg = wrap_in_embed(`Please send responses in the thread created for the issue and not directly in the thread channel.`)
```

Comment:
Maybe just pull the common string out and concatenate the thread link text if it's there? Duplication and all

Reply:
Added a commit to reduce duplication